src/BuildingRegistry.Projections.Wfs/BuildingUnit/BuildingUnitProjections.cs

```diff
@@ -0,0 +1,467 @@
+namespace BuildingRegistry.Projections.Wfs.BuildingUnit
+{
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore;
+    using Be.Vlaanderen.Basisregisters.Utilities.HexByteConvertor;
+    using BuildingRegistry.Building.Events;
+    using Infrastructure;
+    using NetTopologySuite.IO;
+    using NodaTime;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Be.Vlaanderen.Basisregisters.GrAr.Legacy;
+    using Be.Vlaanderen.Basisregisters.GrAr.Legacy.Gebouweenheid;
+    using BuildingRegistry.Building.Events.Crab;
+    using NetTopologySuite.Geometries;
+    using ValueObjects;
+
+    public class BuildingUnitProjections : ConnectedProjection<WfsContext>
+    {
+        private readonly WKBReader _wkbReader;
+
+        public BuildingUnitProjections(WKBReader wkbReader)
+        {
+            _wkbReader = wkbReader;
+
+            #region Building
+            When<Envelope<BuildingPersistentLocalIdWasAssigned>>(async (context, message, ct) =>
+            {
+                var building =
+                    await context.BuildingUnitsBuildings.FindAsync(message.Message.BuildingId, cancellationToken: ct);
+
+                building.BuildingPersistentLocalId = message.Message.PersistentLocalId;
+
+                foreach (var unit in await context.BuildingUnits.GetByBuildingId(message.Message.BuildingId, ct))
+                {
+                    unit.BuildingPersistentLocalId = int.Parse(message.Message.PersistentLocalId.ToString());
+                }
+            });
+
+            When<Envelope<BuildingWasRemoved>>(async (context, message, ct) =>
+            {
+                var buildingUnits = await context.BuildingUnits.GetByBuildingId(message.Message.BuildingId, ct);
+                var unitsToRemove = new List<BuildingUnit>();
+                foreach (var buildingUnitDetailItem in buildingUnits)
+                {
+                    if (message.Message.BuildingUnitIds.Contains(buildingUnitDetailItem.BuildingUnitId))
+                        unitsToRemove.Add(buildingUnitDetailItem);
+                }
+
+                var building = await context.BuildingUnitsBuildings.FindAsync(message.Message.BuildingId, cancellationToken: ct);
+                building.IsRemoved = true;
+
+                context.BuildingUnits.RemoveRange(unitsToRemove);
```

Comment:
Hard delete? This implies they can never be reactivated, or no weird CRAB changes happen to them later